std/encoding.d

```diff
@@ -3380,3 +3380,111 @@ version(unittest)
         return "0123456789ABCDEF"[n & 0xF];
     }
 }
+
+import std.typecons : tuple, Tuple;
+import std.conv : to;
+
+/** Definitions of common Byte Order Marks
+*/
+enum BOM
+{
+    noBOM     = 0,  /// no BOM was found
+    utf32be   = 1,  /// [0x00, 0x00, 0xFE, 0xFF]
+    utf32le   = 2,  /// [0xFF, 0xFE, 0x00, 0x00]
+    utf7      = 3,  /*  [0x2B, 0x2F, 0x76, 0x38]
+                        [0x2B, 0x2F, 0x76, 0x39],
+                        [0x2B, 0x2F, 0x76, 0x2B],
+                        [0x2B, 0x2F, 0x76, 0x2F],
+                        [0x2B, 0x2F, 0x76, 0x38, 0x2D]
```

Comment:
This basically means, you can't skip the bytes if it's UTF 7. I'm hesitant to say this is a blocking criticism, as I think even Robert doesn't know how this particular use case will be handled by user code. But it sure seems wasteful to check for an identical sequence twice

For instance, is that `2d` special? Why would your code want to ignore it? Code that is using this generically can't just say "oh, skip all the bytes that are identified as BOM" because utf7 doesn't allow that. I don't see how this can fit into an algorithm. I guess we can leave it and remove that sequence if it makes sense to later.